### PR TITLE
Refactored the FIRRTL generation

### PR DESCRIPTION
**Pull request by sjalander**
Class name changed from MlirGen to RhlsToFirrtlConverter
Two classes have been reduced to a single class

### CODE REVIEW

File: jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.cpp [hunk 1]
```diff
@@ -1937,85 +1960,83 @@ MLIRGenImpl::MlirGen(const llvm::lambda::node * lambdaNode)
...
-MLIRGenImpl::AddClockPort(::llvm::SmallVector<circt::firrtl::PortInfo> * ports)
+RhlsToFirrtlConverter::AddClockPort(::llvm::SmallVector<circt::firrtl::PortInfo> * ports)
 {
   struct circt::firrtl::PortInfo port = {
-    builder.getStringAttr("clk"),
-    circt::firrtl::ClockType::get(builder.getContext()),
-    circt::firrtl::Direction::In,
-    {},
-    location,
+    Builder_->getStringAttr("clk"), circt::firrtl::ClockType::get(Builder_->getContext()),
```
Comment (phate) [inline jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.cpp:1966]: Each argument on a single line.

File: jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.cpp [hunk 2]
```diff
@@ -1937,85 +1960,83 @@ MLIRGenImpl::MlirGen(const llvm::lambda::node * lambdaNode)
...
-MLIRGenImpl::AddResetPort(::llvm::SmallVector<circt::firrtl::PortInfo> * ports)
+RhlsToFirrtlConverter::AddResetPort(::llvm::SmallVector<circt::firrtl::PortInfo> * ports)
 {
   struct circt::firrtl::PortInfo port = {
-    builder.getStringAttr("reset"),
-    circt::firrtl::IntType::get(builder.getContext(), false, 1),
-    circt::firrtl::Direction::In,
-    {},
-    location,
+    Builder_->getStringAttr("reset"), circt::firrtl::IntType::get(Builder_->getContext(), false, 1),
```
Comment (phate) [inline jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.cpp:1978]: Each argument on a single line

File: jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.cpp [hunk 3]
```diff
@@ -1937,85 +1960,83 @@ MLIRGenImpl::MlirGen(const llvm::lambda::node * lambdaNode)
...
+      Builder_->getStringAttr("width"),
       false,
-      circt::firrtl::IntType::get(builder.getContext(), false, 3)));
+      circt::firrtl::IntType::get(Builder_->getContext(), false, 3)));
 
-  auto memType = circt::firrtl::BundleType::get(builder.getContext(), memReqElements);
+  auto memType = circt::firrtl::BundleType::get(Builder_->getContext(), memReqElements);
   struct circt::firrtl::PortInfo memBundle = {
-    builder.getStringAttr("mem_req"), memType, circt::firrtl::Direction::Out, {}, location,
+    Builder_->getStringAttr("mem_req"), memType, circt::firrtl::Direction::Out, {},
```
Comment (phate) [inline jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.cpp:2012]: You are mixing here. Either each argument on a new line, or all of them on a single line. First option is preferred.

File: jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.hpp [hunk 4]
```diff
@@ -58,16 +61,9 @@ class MLIRGenImpl : public BaseHLS
   void
   WriteCircuitToFile(const circt::firrtl::CircuitOp circuit, std::string name);
   std::string
-  toString(const circt::firrtl::CircuitOp circuit);
+  ToString(llvm::RvsdgModule & rvsdgModule);
```
Comment (phate) [inline jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.hpp:64]: A blank line in between the two functions.

**Review by phate — APPROVED** [review]
Some minor layout comments.

File: jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.cpp (hunk 1 above)
Comment (sjalander) [reply jlm/hls/backend/rhls2firrtl/RhlsToFirrtlConverter.cpp]: It's clang format that places them like this

**Review by phate — APPROVED** [review]
Reversed the formatting myself again. The layout enforced by clang-format is not great. I created a work item on it. See #394

**Pull request merged**